chips/lowrisc/src/pinmux.rs

```diff
@@ -0,0 +1,851 @@
+  Iob9 = 20,
+  Iob10 = 21,
+  Iob11 = 22,
+  Iob12 = 23,
+  Ioc0 = 24,
+  Ioc1 = 25,
+  Ioc2 = 26,
+  Ioc3 = 27,
+  Ioc4 = 28,
+  Ioc5 = 29,
+  Ioc6 = 30,
+  Ioc7 = 31,
+  Ioc8 = 32,
+  Ioc9 = 33,
+  Ioc10 = 34,
+  Ioc11 = 35,
+  Ioc12 = 36,
+  Ior0 = 37,
+  Ior1 = 38,
+  Ior2 = 39,
+  Ior3 = 40,
+  Ior4 = 41,
+  Ior5 = 42,
+  Ior6 = 43,
+  Ior7 = 44,
+  //Ior8
+  //Ior9
+  Ior10 = 45,
+  Ior11 = 46,
+  Ior12 = 47,
+  Ior13 = 48,
+}
+
+impl TryFrom<u8> for MioPadInput
+{
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(MioPadInput::ConstantZero),
+            1 => Ok(MioPadInput::ConstantOne),
+            2 => Ok(MioPadInput::Ioa0),
+            3 => Ok(MioPadInput::Ioa1),
+            4 => Ok(MioPadInput::Ioa2),
+            5 => Ok(MioPadInput::Ioa3),
+            6 => Ok(MioPadInput::Ioa4),
+            7 => Ok(MioPadInput::Ioa5),
+            8 => Ok(MioPadInput::Ioa6),
+            9 => Ok(MioPadInput::Ioa7),
+            10 => Ok(MioPadInput::Ioa8),
+            11 => Ok(MioPadInput::Iob0),
+            12 => Ok(MioPadInput::Iob1),
+            13 => Ok(MioPadInput::Iob2),
+            14 => Ok(MioPadInput::Iob3),
+            15 => Ok(MioPadInput::Iob4),
+            16 => Ok(MioPadInput::Iob5),
+            17 => Ok(MioPadInput::Iob6),
+            18 => Ok(MioPadInput::Iob7),
+            19 => Ok(MioPadInput::Iob8),
+            20 => Ok(MioPadInput::Iob9),
+            21 => Ok(MioPadInput::Iob10),
+            22 => Ok(MioPadInput::Iob11),
+            23 => Ok(MioPadInput::Iob12),
+            24 => Ok(MioPadInput::Ioc0),
+            25 => Ok(MioPadInput::Ioc1),
+            26 => Ok(MioPadInput::Ioc2),
+            27 => Ok(MioPadInput::Ioc3),
+            28 => Ok(MioPadInput::Ioc4),
+            29 => Ok(MioPadInput::Ioc5),
+            30 => Ok(MioPadInput::Ioc6),
+            31 => Ok(MioPadInput::Ioc7),
+            32 => Ok(MioPadInput::Ioc8),
+            33 => Ok(MioPadInput::Ioc9),
+            34 => Ok(MioPadInput::Ioc10),
+            35 => Ok(MioPadInput::Ioc11),
+            36 => Ok(MioPadInput::Ioc12),
+            37 => Ok(MioPadInput::Ior0),
+            38 => Ok(MioPadInput::Ior1),
+            39 => Ok(MioPadInput::Ior2),
+            40 => Ok(MioPadInput::Ior3),
+            41 => Ok(MioPadInput::Ior4),
+            42 => Ok(MioPadInput::Ior5),
+            43 => Ok(MioPadInput::Ior6),
+            44 => Ok(MioPadInput::Ior7),
+            //Ior8
+            //Ior9
+            //Ior10
+            //Ior11
+            45 => Ok(MioPadInput::Ior12),
+            46 => Ok(MioPadInput::Ior11),
+            47 => Ok(MioPadInput::Ior12),
+            48 => Ok(MioPadInput::Ior13),
+            _ => Err(()),
+        }
+    }
+}
+
+impl MioPeripheralInput {
+    pub fn is_lock(&self, reg: &StaticRef<PinmuxRegisters>) -> bool
+    {
+        !reg.mio_periph_insel_regwen[*self as usize].is_set(MIO_PERIPH_INSEL_REGWEN::EN_0)
+    }
```

Comment:
It should be possibile to make this Pinmux configuration full stateless.
I need to find a way to remove "reg: &StaticRef<PinmuxRegisters>)" and replace it with static or const.